### ISSUE TITLE
[MD] The Curse of Illmoore Bay Demo (Unlicensed): Graphical bugs and freeze on map screen.

### ISSUE THREAD

**Issue by BiZkViT96** [higan-emu/ares#1077]
**Describe the issue**
Game has some graphical bugs in options and on map screen. When you get onto the map screen, game will stop responding to the controls.

![2023-04-10 21_39_25-illmoore_demo](https://user-images.githubusercontent.com/88586431/230923792-5ab3025f-9431-4b7a-855e-9be4c18ce926.png)
![2023-04-10 21_39_50-illmoore_demo](https://user-images.githubusercontent.com/88586431/230923794-ed207675-f6eb-47b6-9b0d-34873bd33db4.png)

Work fine in BlastEm.

![2023-04-10 21_42_41-THE CURSE OF ILLMOORE BAY - BlastEm - 59 8 fps](https://user-images.githubusercontent.com/88586431/230924379-63e62bd3-fc9c-4988-a8a6-171984e028f9.png)
![2023-04-10 21_42_58-THE CURSE OF ILLMOORE BAY - BlastEm - 59 9 fps](https://user-images.githubusercontent.com/88586431/230924387-9732f221-0c6a-4e8c-b1d4-22455013cc19.png)
![2023-04-10 21_43_19-THE CURSE OF ILLMOORE BAY - BlastEm - 59 9 fps](https://user-images.githubusercontent.com/88586431/230924389-6d6107bf-388c-4b08-b888-059251d3071a.png)

**Rom Information**

```
file: The Curse of Illmoore Bay Demo
sha256: b4720be68720d82143825a711e73191bfa4321fc90572ce1e63ee126eb684d5a
```
Demo rom https://www.second-dimension.com/downloads/illmoore

ares nightly 2023-04-10

**Comment by TascoDLX** [comment]
There's a subtle bug in the game's boot code which causes it to skip initialization on cold boot, thus not clearing workram. Note, ares inits workram with 0xff, while BlastEm (by default) inits workram with 0x00, thereby masking the issue.

**Comment by BiZkViT96** [comment]
With fix at 000002A6: 4A79

![2023-04-11 11_33_52-The Curse of Illmoore Bay Demo](https://user-images.githubusercontent.com/88586431/231057411-7df13bf1-acb3-4be1-88c6-e5a8eb4ac9df.png)
![2023-04-11 11_34_11-The Curse of Illmoore Bay Demo](https://user-images.githubusercontent.com/88586431/231057416-224bbf26-dffb-4ceb-901a-764f353f64f9.png)


Game bug.
